LFS-749: Non-functioning parent links on the chart page

marta- opened the pull request.

marta- committed:
LFS-749: Non-functioning parent links on the chart page
Forcing the whole page to redraw via history.go(0) when clicking on a parent link.
Downsides:
* _Everything_ is rerendered, including side bar
* Now the page layout is a bit off due to moving parentDetails and needs restiling

marta- committed:
LFS-749: Non-functioning parent links on the chart page
Update the id in the root component when clicking a navigation link

marta- committed:
[cleanup] removed unused code

sashaandjic commented:
The original problem is fixed, however pressing on the back button renders nothing (the url changes though). On the second click on the Back bttn you go 2 steps back as expected.

marta- committed:
LFS-749: Non-functioning parent links on the chart page
Updated to listen to location change, extracted the id using the same logic as in `Subjects.jsx`.

marta- committed:
Merge branch 'dev' into LFS-749-experiment

marta- commented:
@sashaandjic the back button issue should now be fixed.

marta- committed:
[misc] Moved effect that updates the page title to the top component

marta- committed:
LFS-749: Non-functioning parent links on the chart page
Avoid duplication of Subject path regexp

sashaandjic reviewed (approved):
tested
